File: MyLoftCoint/app/src/main/java/com/android/hootr/myloftcoint/di/module/PrefsModule.java
package com.android.hootr.myloftcoint.di.module;

import android.app.Application;

import com.android.hootr.myloftcoint.App;
import com.android.hootr.myloftcoint.data.prefs.Prefs;
import com.android.hootr.myloftcoint.data.prefs.PrefsImpl;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class PrefsModule {

    @Provides
    @Singleton
    public Prefs providePrefs() {
        return new PrefsImpl((Application) App.getInstance());
    }
}

File: TSD/app/src/main/java/com/android/hootr/tsd/di/module/PrefsModule.kt
package com.android.hootr.tsd.di.module

import android.app.Application
import com.android.hootr.tsd.prefs.PrefsImp
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module(includes = [ApplicationModule::class])
class PrefsModule {

    @Singleton
    @Provides
    fun providePrefs(application: Application): PrefsImp {
        return PrefsImp(application.applicationContext)
    }
}